allen-cell-animated/nucmorph-colorizer
# feat: Disabling feature thresholds

Problem
=======
Partially resolves action items in #124

Solution
========
- Allows thresholds on features to persist when switching datasets, even if the feature doesn't exist in the new dataset.
  - Features that are not available are greyed out/disabled.
- Various improvements to formatting on the Select component in `FeatureThresholdingPanel.tsx`
  - Fixed a spacing issue for selected feature names
  - Replaced dropdown and search icons to be consistent with the rest of the UI
- Various improvements to Slider formatting in `LabeledRangeSlider.tsx`
  - Greyed out min/max slider bound text labels when disabled. 
  - Removed tooltips when disabled.

## Type of change

* Bug fix (non-breaking change which fixes an issue)
* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Open the preview link
2. Load this URL: `https://dev-aics-dtp-001.int.allencell.org/dan-data/colorizer/data/test/`
3. Set a threshold on the feature `feature`.
4. Switch to another dataset (Test 5 or Test 6)
5. See that the original feature threshold is now disabled
6. Add additional feature thresholds.
7. Switch back.  All thresholds should persist

NEW: Verifying that thresholds are differentiated by feature + unit
1. Open the preview link. `Test 1 Default` should have a feature with a unit (`feature (test unit)`). If you don't see the unit label, you may need to clear your cache. 
3. Add a filter for that feature.
4. Switch to dataset `Test 2 Nested`. The original feature (`feature (test unit)`) should be greyed out.
5. You should now be able to add a new filter for the feature `feature`.

Demo Video (turn on sound 🔊)
-----------------------

https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/867de740-8d79-42ec-8175-7c0daa43ea99
